tests/test_frbc_tunes.py

```diff
@@ -0,0 +1,272 @@
+    await tasks["send_fill_level_target_profile"]
+
+    start = datetime(2024, 1, 1, 0, 0, tzinfo=timezone(timedelta(seconds=3600)))
+
+    first_call = fm_client.post_measurements.call_args_list[0][1]
+    assert first_call["sensor_id"] == 2
+    assert first_call["start"] == start
+    assert np.isclose(first_call["values"].values, [0] * 4 + [10] * 8 + [20] * 12).all()
+
+    second_call = fm_client.post_measurements.call_args_list[1][1]
+    assert second_call["sensor_id"] == 3
+    assert second_call["start"] == start
+    assert np.isclose(
+        second_call["values"].values, [100] * 4 + [90] * 8 + [80] * 12
+    ).all()
+
+    await cem.close()
+    get_pending_tasks()
+
+
+@pytest.mark.asyncio
+async def test_fill_rate_relay(cem_in_frbc_control_type):
+    cem, fm_client = await cem_in_frbc_control_type
+    frbc = cem._control_types_handlers[cem.control_type]
+
+    actuator_status = {
+        "active_operation_mode_id": "tarnoc-operation-mode",
+        "actuator_id": "id-of-the-actuator",
+        "message_type": "FRBC.ActuatorStatus",
+        "message_id": "a-valid-id",
+        "operation_mode_factor": 0.0,
+    }
+
+    await cem.handle_message(actuator_status)
+
+    tasks = get_pending_tasks()
+
+    # clear mock state because it contains previous such as
+    # the ones used to process the system description
+    fm_client.reset_mock()
+
+    # wait for the task send_actuator_status to finish
+    await tasks["send_actuator_status"]
+
+    first_call = fm_client.post_measurements.call_args_list[0][1]
+    assert first_call["sensor_id"] == frbc._thp_fill_rate_sensor_id
+
+    second_call = fm_client.post_measurements.call_args_list[1][1]
+    assert second_call["sensor_id"] == frbc._fill_rate_sensor_id
+
+    # Switch operation mode to Nestore
+    actuator_status["active_operation_mode_id"] = "nestore-operation-mode"
+
+    await cem.handle_message(actuator_status)
+    tasks = get_pending_tasks()
+
+    # clear mock state because it contains previous such as
+    # the ones used to process the system description
+    fm_client.reset_mock()
+
+    # wait for the task send_actuator_status to finish
+    await tasks["send_actuator_status"]
+
+    first_call = fm_client.post_measurements.call_args_list[0][1]
+    assert first_call["sensor_id"] == frbc._nes_fill_rate_sensor_id
+
+    second_call = fm_client.post_measurements.call_args_list[1][1]
+    assert second_call["sensor_id"] == frbc._fill_rate_sensor_id
+
+    await cem.close()
+    get_pending_tasks()
+
+
+@pytest.mark.asyncio
+async def test_trigger_schedule(cem_in_frbc_control_type):
+    cem, fm_client = await cem_in_frbc_control_type
+    # frbc = cem._control_types_handlers[cem.control_type]
+
+    tasks = get_pending_tasks()
+
+    assert tasks["trigger_schedule_task"]._state == "PENDING"
+    await cem.close()
+
+    tasks = get_pending_tasks()
+
+    assert tasks["trigger_schedule_task"]._state == "PENDING"
+
+    await cem.close()
+    get_pending_tasks()
```

Comment:
This test I don't understand.

Reply:
This test checks whether the task starts when the FRBC control type is selected.

Perhaps the task should actually stop when the CEM closes. But at least the RM doesn't get messages anymore after the CEM closes.